pkg/controller/job/job_controller_test.go

```diff
@@ -357,9 +357,9 @@ func TestControllerSyncJob(t *testing.T) {
 			jobPodReplacementPolicy: true,
 			terminatingPods:         1,
 			expectedTerminating:     ptr.To[int32](1),
-			expectedPodPatches:      2,
-			expectedDeletions:       1,
```

Comment:
Explain why we have a difference in `expectedPodPatches` here.

Reply:
I don't really understand what this actually does. Care to explain?

Reply:
Do you want it as a comment?

Researching this, this makes sense.  

We are tracking the number of patches to a pod.  Since we removed the terminating count, we do not include terminating pods in `podsToDelete`.  So since we drop terminating pods we only have the number of active pods that we delete.  

So I think 1 less pod patch makes sense to me.

Reply:
I think it is enough to leave it as a comment here.